.github/workflows/auto-update-contributors-readme.yml

```diff
@@ -0,0 +1,99 @@
+# GitHub Actions documentation:
+# https://docs.github.com/en/actions
+
+name: Auto-update list of contributors
+
+on:
+  workflow_call:
+
+jobs:
+  update_contributors_list:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+
+      - name: Check if the default branch is protected
+        id: default_branch
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          defaultBranch="$( \
+            gh api /repos/${{ github.repository }} \
+              --method GET \
+              --jq '.default_branch'
+          )"
+
+          isProtected="$( \
+            gh api /repos/${{ github.repository }}/branches/$defaultBranch \
+            --jq '.protected'
+          )"
+
+          echo "::set-output name=is_protected::$isProtected"
+
+      - name: Check if a pull request already exists
+        if: ${{ steps.default_branch.outputs.is_protected == 'true' }}
+        id: find_existing_pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ID_PR="$( \
+            gh api search/issues \
+              --method GET \
+              --raw-field q='repo:${{ github.repository }} type:pr state:open "Auto-update list of contributors" in:title' \
+              --jq '.items.[0].number' \
```

Comment:
- https://cli.github.com/manual/gh_api
- https://docs.github.com/en/rest/reference/search#search-issues-and-pull-requests

Reply:
Regarding the filter query:

```
q='repo:${{ github.repository }} type:pr state:open "Auto-update list of contributors" in:title'
```

It is easy to try to build such a query via GitHub, and values do not need to be escaped, for instance:
https://github.com/pulls?q=repo:peopledoc/ember-cli-embedded%20type:pr%20state:closed%20%22release%22%20in:title

Plus, I could not find a better query to ensure to get the right pull request. Filtering on the title seems to be the most reliable way 🤔  . I could eventually add a filter on the author `app/github-actions`.

Reply:
Filter on author added.